File: cpp-answers/228.汇总区间.cpp
/*
 * @Description: 
 * @Author: Hongyang_Yang
 * @Date: 2020-11-14 19:29:48
 * @LastEditors: Hongyang_Yang
 * @LastEditTime: 2020-11-14 19:32:43
 */
/*
 * @lc app=leetcode.cn id=228 lang=cpp
 *
 * [228] 汇总区间
 */

// @lc code=start
class Solution
{
public:
    vector<string> summaryRanges(vector<int> &nums)
    {
        vector<string> res;
        for (int i = 0; i < nums.size(); i++)
        {
            int pre = nums[i];
            while (i + 1 < nums.size() && nums[i + 1] == nums[i] + 1)
                i++;
            if (pre == nums[i])
                res.push_back(to_string(pre));
            else
            {
                res.push_back(to_string(pre) + "->" + to_string(nums[i]));
            }
        }
        return res;
    }
};
// @lc code=end

File: rust-answers/228.汇总区间.rs
/*
 * @lc app=leetcode.cn id=228 lang=rust
 *
 * [228] 汇总区间
 */

// @lc code=start
impl Solution {
    pub fn summary_ranges(nums: Vec<i32>) -> Vec<String> {
        let mut answer = Vec::new();
        let mut i = 0;
        while i < nums.len() {
            let n = nums[i];
            while i < nums.len() - 1 && nums[i + 1] == nums[i] + 1 {
                i += 1;
            }
            let mut s = n.to_string();
            if n != nums[i] {
                s += "->";
                s += &nums[i].to_string();
            }
            answer.push(s);
            i += 1;
        }
        answer
    }
}
// @lc code=end

